secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/exceptions/GlobalExceptionHandler.java

```diff
@@ -105,30 +106,14 @@ protected ResponseEntity<Object> handleServletRequestBindingException(ServletReq
                                                                           HttpHeaders headers,
                                                                           HttpStatus status,
                                                                           WebRequest request) {
-        if (ex.getMessage().startsWith("Missing request header")) {
+        if (ex instanceof MissingRequestHeaderException) {
             return handleOBErrorResponse(
                     new OBErrorResponseException(
                             HttpStatus.BAD_REQUEST,
                             OBRIErrorResponseCategory.REQUEST_INVALID,
                             OBRIErrorType.REQUEST_MISSING_HEADER.toOBError1(ex.getMessage())
                     ),
                     request);
-        } else if (ex.getMessage().startsWith("Missing cookie")) {
-            return handleOBErrorResponse(
-                    new OBErrorResponseException(
-                            HttpStatus.BAD_REQUEST,
-                            OBRIErrorResponseCategory.REQUEST_INVALID,
-                            OBRIErrorType.REQUEST_MISSING_COOKIE.toOBError1(ex.getMessage())
-                    ),
-                    request);
-        } else if (ex.getMessage().startsWith("Missing argument")) {
```

Comment:
Branch removed as not reachable. I've searched the Spring code and there are not exceptions that contain this String, I believe this logic is trying to handle the case where a required API param is missing, this will be handled by the following method instead: `handleMissingServletRequestParameter`